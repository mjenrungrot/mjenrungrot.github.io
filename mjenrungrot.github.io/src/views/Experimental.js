// @flow

// eslint-disable-next-line no-unused-vars
import React from 'react';
// eslint-disable-next-line no-unused-vars
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';

import {withStyles} from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';

import Main from '../layouts/Main';

const styles = (theme) => ({
  root: {
    ...theme.mixins.gutters(),
    paddingTop: theme.spacing(2),
    paddingBottom: theme.spacing(2),
  },
});

/**
 * Experimental View Component
 * @param {object} props creation params
 * @return {object} Experimental component
 */
function Experimental(props) {
  const {classes} = props;
  return (
    <Main>
      <Paper className={classes.root}>
        <Typography variant="h5" component="h3" gutterBottom>
          Experimental
        </Typography>
        <Typography component="p" gutterBottom>
          This component is an experimental section. It&apos;s used for
          developing purposes only.
        </Typography>
      </Paper>
    </Main>
  );
}

Experimental.propTypes = {
  classes: PropTypes.any.isRequired,
};

export default withStyles(styles)(Experimental);
