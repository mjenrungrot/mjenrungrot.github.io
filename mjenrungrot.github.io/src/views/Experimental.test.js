// eslint-disable-next-line no-unused-vars
import React from 'react';
import {shallow} from 'enzyme';
import Experimental from './Experimental';

describe('Experimental component', () => {
  test('renders', () => {
    const component = shallow(<Experimental />);
    expect(component).toMatchSnapshot();
  });
});
