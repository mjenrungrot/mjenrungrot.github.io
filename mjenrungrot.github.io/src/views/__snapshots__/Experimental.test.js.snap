// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Experimental component renders 1`] = `
<Experimental
  classes={
    Object {
      "root": "Experimental-root-1",
    }
  }
/>
`;
